packages/block-library/src/playlist/edit.js

```diff
@@ -0,0 +1,214 @@
+/**
+ * External Dependencies
+ */
+import { uniqBy } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	FormFileUpload,
+	IconButton,
+	Placeholder,
+	ServerSideRender,
+	Toolbar,
+	withNotices,
+	CheckboxControl,
+	PanelBody,
+	SelectControl,
+} from '@wordpress/components';
+import { Component, Fragment } from '@wordpress/element';
+import {
+	BlockControls,
+	InspectorControls,
+	mediaUpload,
+	MediaUpload,
+} from '@wordpress/editor';
+
+class PlaylistEdit extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.initializePlaylist = this.initializePlaylist.bind( this );
+		this.uploadFromFiles = this.uploadFromFiles.bind( this );
+		this.onUploadFiles = this.onUploadFiles.bind( this );
+		this.onSelectMedia = this.onSelectMedia.bind( this );
+		this.getMimeBaseType = this.getMimeBaseType.bind( this );
+	}
+
+	/**
+	* The function is used after the editor has enqueued playlist dependencies.
+	*
+	* @param  {function} initializePlaylist Function that Initializes playlist
+	*/
+
+	initializePlaylist() {
+		window.wp.playlist.initialize();
+	}
+
+	uploadFromFiles( event ) {
+		const { noticeOperations } = this.props;
+		if ( ! event.target.files ) {
+			noticeOperations.createErrorNotice( 'Files list provided is empty' );
+			return;
+		}
+		this.onUploadFiles( event.target.files );
+	}
+
+	getMimeBaseType( type ) {
+		const { noticeOperations } = this.props;
+		if ( ! type ) {
+			noticeOperations.createErrorNotice( 'MIME Type could not be determined' );
+			return;
+		}
+		return type.split( '/' )[ 0 ];
+	}
+
+	onUploadFiles( files ) {
+		const { setAttributes, noticeOperations } = this.props;
+		const isConsistentType = uniqBy( files, ( file ) => this.getMimeBaseType( file.type ) ).length === 1;
+		if ( ! isConsistentType ) {
+			noticeOperations.createErrorNotice( 'Cannot have mixed types in a Playlist Block' );
+			setAttributes( { ids: undefined, type: undefined } );
+			return;
+		}
+
+		mediaUpload( {
+			allowedType: [ 'audio', 'video' ],
+			filesList: files,
+			onFileChange: ( media ) => {
+				// validate type is consistent for playlist
+				if ( media.length > 0 && media[ 0 ].mimeType && isConsistentType ) {
+					const type = media[ 0 ].mimeType.split( '/' )[ 0 ];
+					const ids = media.map( ( item ) => item.id );
+					setAttributes( { ids, type } );
+				}
+			},
+		} );
+	}
+
+	toggleAttribute( attribute ) {
+		return ( newValue ) => {
+			this.props.setAttributes( { [ attribute ]: newValue } );
+		};
+	}
+
+	onSelectMedia( media ) {
+		const { setAttributes } = this.props;
+		//check if there are returned media items and set attributes when there are
+		if ( media && media[ 0 ].url ) {
```

Comment:
I don't believe it will ever be unless there is an error maybe. We also have a similar check in the Audio block. I was matching that for consistency on the chance there was an instance I was unaware of: https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/audio/edit.js#L88